### ISSUE TITLE
[FR] Sanity check when applying M203 settings

### ISSUE THREAD

**Issue by bungee-** [MarlinFirmware/Marlin#19882]
### Description

I had problem with my printer when I tried to print a part but printer rebooted when G28 came. It was not a bug but probably my misconfiguration of slicer software. I use PrusaSlicer and in the G-code slicer sends acceleration and speed settings to the printer. Accidentally between those settings was maximum limit for Z axis set on 0. Obviously settings were accepted and when G28 came Z-axis was not moving as max speed was set at 0. In one iteration of my testing printer did not reset, but software froze and it almost had thermal runaway (luckily I was close). 

TL:DR; Proposing check for M203 values so 0 is not allowed or it's ignored. 

### Feature Workflow

-------------- Option with user warning --------------
1. Stop motors and heaters.
2. Beep
3. Inform the user of faulty G-code setting 
---------------------------------------------------------

------------- Option with ignoring values ------------
1. If value is 0 ignore it and use printer default value
---------------------------------------------------------
